scripts/functions.js

```diff
@@ -0,0 +1,2704 @@
+};
+
+/****************************************************
+ Private helpers
+ ****************************************************/
+
+var apikey = config.get("apiKey");
+
+var concatQuery = function (url, key, value) {
+    return url + ((!url || url.indexOf('?') < 0) ? '?' : '&') + key + "=" + value;
+}
+
+var checkHttpOptions = function (url, options) {
+    options = options || {};
+    if (!!url) {
+        if (isObject(url)) {
+            // take the 'url' parameter as the options
+            options = url || {};
+        } else {
+            if (!!options.path || !!options.params || !!options.body) {
+                // options contains the http package format
+                options.path = url;
+            } else {
+                // create html package
+                options = {
+                    path: url,
+                    body: options
+                }
+            }
+        }
+    }
+    return options;
+}
+
+var isObject = function (obj) {
+    return !!obj && stringType(obj) === '[object Object]'
+}
+
+var stringType = Function.prototype.call.bind(Object.prototype.toString)
+
+var parse = function (str) {
+    try {
+        if (arguments.length > 1) {
+            var args = arguments[1], i = 0;
+            return str.replace(/(:(?:\w|-)+)/g, () => {
+                if (typeof (args[i]) != 'string') throw new Error('Invalid type of argument: [' + args[i] + '] for url [' + str + '].');
+                return args[i++];
+            });
+        } else {
+            if (str) {
+                return str;
+            }
+            throw new Error('No arguments nor url were received when calling the helper. Please check it\'s definition.');
+        }
+    } catch (err) {
+        sys.logs.error('Some unexpected error happened during the parse of the url for this helper.')
+        throw err;
+    }
+}
+
+/****************************************************
+ Constants
+ ****************************************************/
+
+var server = apiKey.substring(apiKey.lastIndexOf('-') + 1);
+var MAILCHIMP_SERVER = server == null || server.trim() === '' ? 'us1' : server.trim();
+var API_URL = "https://" + MAILCHIMP_SERVER + ".api.mailchimp.com/3.0/";
+
+/****************************************************
+ Configurator
+ ****************************************************/
+
+var Mailchimp = function (options) {
+    options = options || {};
+    options= setApiUri(options);
+    options= setRequestHeaders(options);
+    return options;
+}
+
+/****************************************************
+ Private API
+ ****************************************************/
+
+function setApiUri(options) {
+    var url = options.path || "";
+    options.url = API_URL + url;
+    sys.logs.debug('[mailchimp] Set url: ' + options.path + "->" + options.url);
+    return options;
+}
+
+function setRequestHeaders(options) {
+    var headers = options.headers || {};
+    if (config.get("authenticationMethod") === "apiKey") {
```

Comment:
All this if is unnecessary for this package, it is standard code that can be replaced as required by the package configuration.